messages/teleporter/message_manager.go

```diff
@@ -91,21 +91,27 @@ func isAllowedRelayer(allowedRelayers []common.Address, eoa common.Address) bool
 }
 
 // ShouldSendMessage returns true if the message should be sent to the destination chain
-func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageInfo, destinationChainID ids.ID) (bool, error) {
+func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageInfo) (bool, ids.ID, error) {
```

Comment:
Rather than returning the destination ID in this method, I think we should add a separate method to `MessageManager` called `GetDestinationID` that returns an `ids.ID`. That would make more sense logically, and would make this interface more general for future use cases such as anycasting. For the anycast case, we can generalize `GetDestinationID` to something like `GetDestinations`, but I think just adding `GetDestinationID` is fine for now.

Reply:
We could also move the call to `messageManager.GetDestinationID()` to `RelayMessage()`, then pass in the `destinationChainID` to `NewMessageRelayer`. That way we can still compose the message relayer with the `destinationChainID`, instead of having to pass the non-member variable around the function calls like you do in this PR.

Reply:
fixed